### PR TITLE
Class list

### PR DESCRIPTION
**Pull request by koodimetsa**
This is an implementation of classList.

### CODE REVIEW

File: classList.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package dom
+
+import "github.com/dennwc/dom/js"
+
+type ClassList struct {
```
Comment (dennwc) [inline classList.go]: According to [docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) it should be `dom.TokenList`.

Comment (dennwc) [inline classList.go]: Also, please implement the `As***` function, so users can wrap other values into this type.

File: node.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ type Node interface {
 
 	AppendChild(n Node)
 	AttachShadow() Node
+	ClassList() *ClassList
```
Comment (dennwc) [inline node.go]: According to [docs](https://developer.mozilla.org/en-US/docs/Web/API/Element) this should be a function on `dom.Element`, not `dom.Node`.

**Comment by koodimetsa** [comment]
Hi, thank you for guidance. Next time I will check the specification more carefully.

I think I got the other changes, but I'm not sure about the As-func.

Would something like this be good:
```
func AsTokenList(v js.Value) *TokenList {
	if !v.Valid() {
		return nil
	}
	return &TokenList{v: v}
}
```

**Comment by dennwc** [comment]
@koodimetsa Sure, just like other functions in this package.

File: element.go [hunk 3]
```diff
@@ -76,3 +76,7 @@ func (e *Element) OnMouseMove(h MouseEventHandler) {
 func (e *Element) OnMouseUp(h MouseEventHandler) {
 	e.onMouseEvent("mouseup", int(sjs.StopPropagation), h)
 }
+
+func (e *Element) ClassList() *TokenList {
+	return &TokenList{v: e.v.Get("classList")}
```
Comment (dennwc) [inline element.go]: Please use `AsTokenList` here

Comment (koodimetsa) [reply element.go]: Hi, I added a fix.

**Pull request merged**